Foundation/URLSession/URLSession.swift

```diff
@@ -192,6 +192,7 @@ public let NSURLSessionTransferSizeUnknown: Int64 = -1
 open class URLSession : NSObject {
     fileprivate let _configuration: _Configuration
     fileprivate let multiHandle: _MultiHandle
+    fileprivate let taskIdentifierLock = NSLock()
```

Comment:
my only question is what is the preferred way, `NSLock` or a dispatch queue? I think we should be consistent. @phausler / @parkera do we have a preferred option?

Reply:
I’ve switched to using the workQueue for consistency